pkg/autoscaler/autoscaler.go

```diff
@@ -401,13 +360,28 @@ func (a *Autoscaler) targetConcurrency() float64 {
 	return a.target
 }
 
-func (a *Autoscaler) podCountLimited(desiredPodCount, currentPodCount float64) float64 {
-	return math.Min(desiredPodCount, a.Current().MaxScaleUpRate*currentPodCount)
+func (a *Autoscaler) podCountLimited(desiredPodCount float64, currentPodCount int) float64 {
+	// Use 1 as minimum for current pods.
+	return math.Min(desiredPodCount, a.Current().MaxScaleUpRate*math.Max(1, float64(currentPodCount)))
+}
+
+func isActivator(podName string) bool {
+	// TODO(#2282): This can cause naming collisions.
+	return strings.HasPrefix(podName, ActivatorPodName)
+}
+
+func divide(a, b float64) float64 {
+	if math.Abs(b) < approximateZero {
+		return 0
+	}
+	return a / b
 }
 
-func (a *Autoscaler) readyPods() (float64, error) {
+// readyPodsOfEndpoints returns the ready IP count in the K8S Endpoints object returned by
+// the given K8S Informer with given namespace and name. This is same as ready Pod count.
+func readyPodsOfEndpoints(lister corev1listers.EndpointsLister, ns, name string) (int, error) {
```

Comment:
can you add a suffix like Count to it, to indicate it is returning the count and not the pods

Reply:
Done.